### PR TITLE
Add OCP metadata to each document

### PR DESCRIPTION
**Pull request by rsevilla87**
### Description

Add clusterMetadata collected by the OCP wrapper to each indexed document. Also adds the flag --user-metadata to the ocp wrapper, index and init entrypoints.

Fixes: https://github.com/cloud-bulldozer/kube-burner/issues/250

### CODE REVIEW

**Comment by jtaleric** [comment]
So I tested this, and I see some odd behavior 
```
DEBU[2023-02-17 21:51:19] Indexing [1] documents in ripsaw-kube-burner                                                                                                                                             
DEBU[2023-02-17 21:51:19] Indexing finished in 7ms: created=1                                                                                                                                                      
DEBU[2023-02-17 21:51:19] Instant query: count(kube_configmap_info{})                                                                                                                                              
DEBU[2023-02-17 21:51:19] Indexing [1] documents in ripsaw-kube-burner                                                                                                                                             
DEBU[2023-02-17 21:51:19] Indexing finished in 7ms: created=1                                                                                                                                                      
DEBU[2023-02-17 21:51:19] Instant query: count(kube_service_info{})                                                                                                                                                
DEBU[2023-02-17 21:51:19] Indexing [1] documents in ripsaw-kube-burner                                                                                                                                             
DEBU[2023-02-17 21:51:19] Indexing finished in 9ms: created=1                                                                                                                                                      
DEBU[2023-02-17 21:51:19] Range query: kube_node_role                                                                                                                                                              
DEBU[2023-02-17 21:51:19] Indexing [264] documents in ripsaw-kube-burner                                                                                                                                           
DEBU[2023-02-17 21:51:19] Indexing finished in 54ms: created=264                                                                                                                                                   
DEBU[2023-02-17 21:51:19] Range query: sum(kube_node_status_condition{status="true"}) by (condition)                                                                                                               
DEBU[2023-02-17 21:51:19] Indexing [32] documents in ripsaw-kube-burner    
```

However, we only see a single document - I wonder if ExtraFields is causing issues in OpenSearch? 
image

**Comment by rsevilla87** [comment]
> So I tested this, and I see some odd behavior
> 
> ```
> DEBU[2023-02-17 21:51:19] Indexing [1] documents in ripsaw-kube-burner                                                                                                                                             
> DEBU[2023-02-17 21:51:19] Indexing finished in 7ms: created=1                                                                                                                                                      
> DEBU[2023-02-17 21:51:19] Instant query: count(kube_configmap_info{})                                                                                                                                              
> DEBU[2023-02-17 21:51:19] Indexing [1] documents in ripsaw-kube-burner                                                                                                                                             
> DEBU[2023-02-17 21:51:19] Indexing finished in 7ms: created=1                                                                                                                                                      
> DEBU[2023-02-17 21:51:19] Instant query: count(kube_service_info{})                                                                                                                                                
> DEBU[2023-02-17 21:51:19] Indexing [1] documents in ripsaw-kube-burner                                                                                                                                             
> DEBU[2023-02-17 21:51:19] Indexing finished in 9ms: created=1                                                                                                                                                      
> DEBU[2023-02-17 21:51:19] Range query: kube_node_role                                                                                                                                                              
> DEBU[2023-02-17 21:51:19] Indexing [264] documents in ripsaw-kube-burner                                                                                                                                           
> DEBU[2023-02-17 21:51:19] Indexing finished in 54ms: created=264                                                                                                                                                   
> DEBU[2023-02-17 21:51:19] Range query: sum(kube_node_status_condition{status="true"}) by (condition)                                                                                                               
> DEBU[2023-02-17 21:51:19] Indexing [32] documents in ripsaw-kube-burner    
> ```
> 
> However, we only see a single document - I wonder if ExtraFields is causing issues in OpenSearch? image

Yeah, the original implementation did not work well, I've updated the code. Example indexed document:

image


Do you miss any other metadata?

File: pkg/workloads/helpers.go [hunk 1]
```diff
@@ -163,6 +163,13 @@ func (wh *WorkloadHelper) indexMetadata() {
 }
 
 func (wh *WorkloadHelper) run(workload, metrics string) {
+	metadata := map[string]interface{}{
```
Comment (jtaleric) [inline pkg/workloads/helpers.go:166]: IMHO these are the main things... 

In the future, we can consider `kernel and arch/cpu type` thoughts?

**Review by jtaleric — APPROVED** [review]
lgtm. We can add additional metadata as we see fit.

File: pkg/workloads/helpers.go (hunk 1 above)
Comment (afcollins) [reply pkg/workloads/helpers.go:166]: We should already have access to the cpu arch. GetNodeInfo already lists all nodes.  the label we can use is `kubernetes.io/arch={arm64,amd64}`. A design decision is whether to index once per cluster or per node. We can expect to have heterogenous clusters eventually, thus I am not sure how we would aggregate into cluster-metadata (would we check whether all controlplane/infra/workers match, and conditionally set an optional field? 🤔  )

**Review by afcollins — APPROVED** [review]
lgtm Thanks!

**Comment by rsevilla87** [comment]
Example with custom provide user-metadata

```
cloud-bulldozer: true
```

image

**Review by jtaleric — APPROVED** [review]
lgtm - still need to test.

**Pull request merged**